src/messaging/CredentialApiMessageType.ts

```diff
@@ -104,8 +104,9 @@ export function ensureOwnerIsSender(message: IMessage): void {
  *
  * @param decryptedMessage The decrypted message to check.
  */
-export function verify(decryptedMessage: IMessage): void {
+export function assertKnownMessage(decryptedMessage: IMessage): IMessage['body'] {
```

Comment:
```suggestion
export function assertKnownMessage(decryptedMessage: IMessage): IMessage<KnownMessageBodies>['body'] {
```
not sure what the union type of all known body types is called now, but this is the point of this function: we assert that the message body is of one of those types.